src/main/java/com/xpanxion/java/assignments/student1/Worker1.java

```diff
@@ -1,8 +1,25 @@
 package com.xpanxion.java.assignments.student1;
 
+import com.xpanxion.java.assignments.DataAccess;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
 public class Worker1 {
 
     public void ex1() {
-        System.out.println("Ex. 1...");
+        var products = DataAccess.getProducts();
+        var departments = DataAccess.getDepartments();
+        products
+                .stream()
+                .map(product -> product)
+                .forEach(p -> {
+                    departments.forEach(d -> {
```

Comment:
This approach has performance implications when the department list becomes very long.  Let's try to convert the department List to a Map, which will result in much faster lookups.

https://howtodoinjava.com/java8/collect-stream-to-map/

Reply:
Got it. Changes have been pushed